src/writeToFileAsync.js

```diff
@@ -0,0 +1,23 @@
+const fs = require('fs').promises;
+
+async function writeToFileAsync(filePath, content) {
+  try {
+    await fs.writeFile(filePath, content);
+    const stats = await fs.stat(filePath);
+    return stats.size / 1024;
+  } catch (error) {
+    return null;
+  }
+}
+
+(async () => {
+  const size = await writeToFileAsync(
+    'example.txt',
+    'Hello, World! \nasync file content included',
+  );
+  if (size !== null) {
+    console.log(`File saved successfully. Size: ${size.toFixed(2)} KB`);
+  } else {
+    console.log('File saving failed.');
+  }
+})();
```

Comment:
You're using an Immediately Invoked Function Expression here (IIFE). It works, but it's not super elegant. I suggest using a regular function if you want to use `async`.
```
async function writeHelloWorldExample() {
  // ...
}
writeHelloWorldExample();
```